### ISSUE TITLE
cjdroute does not create a link under windows 7 64bit

### ISSUE THREAD

**Issue by vikulin** [interfect/cjdns-installer#11]
c:\Users\Vadym\Desktop\cjdns>cjdroute --nobg < cjdroute.conf
1489003472 INFO cjdroute2.c:642 Cjdns i386 win32
1489003472 INFO cjdroute2.c:646 Checking for running instance...
1489003472 DEBUG UDPAddrIface.c:293 Bound to address [0.0.0.0:56103]
1489003472 DEBUG AdminClient.c:333 Connecting to [127.0.0.1:11234]
1489003472 DEBUG Pipe.c:134 Buffering a message
1489003472 DEBUG cjdroute2.c:699 Sent [156] bytes to core
1489003472 INFO RandomSeed.c:42 Attempting to seed random number generator
1489003472 INFO RandomSeed.c:50 Trying random seed [RtlGenRandom() (Windows)] Su
ccess
1489003472 INFO RandomSeed.c:64 Seeding random number generator succeeded with [
1] sources
1489003472 INFO LibuvEntropyProvider.c:59 Taking clock samples every [1000]ms fo
r random generator
11489003472 DEBUG 489003472 DEBUG PCipe.core.c::231 P354 ipe [\\.\pipe\cjdns_pip
e_client-core-vsvukbvpvzq5pt22chpckju8q75fm1] established connection
1489003472 DEBUG Pipe.c:253 GSetting pre-configuration from client
ending buffered message
1489003472 DEBUG Pipe.c:231 Pipe [\\.\pipe\cjdns_pipe_client-core-vsvukbvpvzq5pt
22chpckju8q75fm1] established connection
1489003472 DEBUG Core.c:357 Finished getting pre-configuration from client
1489003472 DEBUG UDPAddrIface.c:254 Binding to address [127.0.0.1:11234]
1489003472 DEBUG UDPAddrIface.c:293 Bound to address [127.0.0.1:11234]
1489003472 DEBUG UDPAddrIface.c:293 Bound to address [0.0.0.0:56104]
1489003472 DEBUG AdminClient.c:333 Connecting to [127.0.0.1:11234]
1489003472 INFO Configurator.c:135 Checking authorized password 0.
11489003472 DEBUG Pathfinder.c:489003472 INFO 165 CIonfigurator.cNIT
:1489003472 DEBUG 159 NAodeStore.cdding authorized password #[0] for user [defau
lt-login].
:565 Linking [fc1e:3253:74fa:0b29:70e9:07b0:e778:aaad] with [fc1e:3253:74fa:0b29
:70e9:07b0:e778:aaad] with label fragment [0000.0000.0000.0001]
1489003472 DEBUG UDPAddrIface.c:254 Binding to address [[::]:44617]
1489003472 DEBUG UDPAddrIface.c:293 Bound to address [[::ffff:0.0.0.0]:44617]
1489003472 DEBUG Core.c:155 Initializing TUN device [<auto>]
1489003472 DEBUG TAPInterface.c:246 Getting TAP-Windows device name
1489003472 DEBUG TAPInterface.c:252 Opening TAP-Windows device [LAN2] at locatio
n [\\.\Global\{22ED0F37-D4B8-4C7B-ABA6-4D0C5C78BE3D}.tap]
1489003472 INFO TAPInterface.c:289 Opened TAP-Windows device [LAN2] version [9.9
.0] at location [\\.\Global\{22ED0F37-D4B8-4C7B-ABA6-4D0C5C78BE3D}.tap]
1489003472 DEBUG TAPInterface.c:135 Posted read
1489003472 INFO NetDev.c:63 Setting IP address [fc1e:3253:74fa:b29:70e9:7b0:e778
:aaad/8/8] on interface [LAN2]
1489003472 DEBUG NetPlatform_win32.c:225 Going to run command: netsh interface i
pv6 set subinterface "LAN2" mtu=1304
ОК.

1489003472 DEBUG NetPlatform_win32.c:234 Going to run command: netsh interface i
pv4 set subinterface "LAN2" mtu=1304
ОК.

11489003472 DEBUG 489003472 DEBUG TCAPInterface.configurator.c::148 531 RSead [3
42] bytes
ecurity_chroot(/var/run/)
1489003472 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003472 DEBUG TAPInterface.c:135 Posted read
1489003472 DEBUG TAPInterface.c:148 Read [78] bytes
1489003472 DEBUG NDPServer.c:115 Sending neighbor advert
1489003472 DEBUG TAPInterface.c:180 Posted write [86] bytes
1489003472 DEBUG TAPInterface.c:135 Posted read
1489003472 DEBUG 1TAPInterface.c:489003472 DEBUG 208 CAonfigurator.cll pending w
rites are complete
:1576 489003472 DEBUG STecurity_noforks()
APInterface.c:148 Read [62] bytes
1489003472 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003472 DEBUG TAPInterface.c:135 Posted read
1489003472 DEBUG Configurator.c:596 Security_seccomp()
1489003472 DEBUG Configurator.c:601 Security_setupComplete()
1489003472 DEBUG Configurator.c:685 Cjdns started in the background
1489003472 DEBUG cjdroute2.c:740 Keeping cjdns client alive because --nobg was s
pecified on the command line
1489003473 DEBUG Janitor.c:636 Could not find anything to do
1489003473 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003473 DEBUG TAPInterface.c:148 Read [90] bytes
1489003473 DEBUG NDPServer.c:54 wrong type/code for neighbor solicitation
1489003473 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003473 DEBUG TAPInterface.c:135 Posted read
1489003473 DEBUG TAPInterface.c:148 Read [95] bytes
1489003473 DEBUG NDPServer.c:54 wrong type/code for neighbor solicitation
1489003473 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003473 DEBUG TAPInterface.c:135 Posted read
1489003474 DEBUG TAPInterface.c:148 Read [90] bytes
1489003474 DEBUG NDPServer.c:54 wrong type/code for neighbor solicitation
1489003474 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003474 DEBUG TAPInterface.c:135 Posted read
1489003474 DEBUG Janitor.c:636 Could not find anything to do
1489003474 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003475 DEBUG TAPInterface.c:148 Read [86] bytes
1489003475 DEBUG NDPServer.c:54 wrong type/code for neighbor solicitation
1489003475 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003475 DEBUG TAPInterface.c:135 Posted read
1489003475 DEBUG TAPInterface.c:148 Read [342] bytes
1489003475 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003475 DEBUG TAPInterface.c:135 Posted read
1489003475 DEBUG Janitor.c:636 Could not find anything to do
1489003475 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003476 DEBUG Janitor.c:636 Could not find anything to do
1489003476 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003476 DEBUG TAPInterface.c:148 Read [62] bytes
1489003476 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003476 DEBUG TAPInterface.c:135 Posted read
1489003476 DEBUG Janitor.c:636 Could not find anything to do
1489003476 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003477 DEBUG Janitor.c:636 Could not find anything to do
1489003477 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003477 DEBUG TAPInterface.c:148 Read [86] bytes
1489003477 DEBUG NDPServer.c:54 wrong type/code for neighbor solicitation
1489003477 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003477 DEBUG TAPInterface.c:135 Posted read
1489003478 DEBUG Janitor.c:636 Could not find anything to do
1489003478 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003479 DEBUG TAPInterface.c:148 Read [42] bytes
1489003479 DEBUG TUNAdapter.c:56 DROP packet unknown ethertype [2054]
1489003479 DEBUG TAPInterface.c:135 Posted read
1489003479 DEBUG Janitor.c:636 Could not find anything to do
1489003479 DEBUG Janitor.c:664 Global Mean Response Time: 5000 nodes [1] links [
1]
1489003480 DEBUG TAPInterface.c:148 Read [42] bytes
1489003480 DEBUG TUNAdapter.c:56 DROP packet unknown ethertype [2054]
1489003480 DEBUG TAPInterface.c:135 Posted read
1489003480 DEBUG TAPInterface.c:148 Read [86] bytes
1489003480 DEBUG NDPServer.c:54 wrong type/code for neighbor solicitation
1489003480 INFO IpTunnel.c:710 Message with unrecognized address from TUN
1489003480 DEBUG TAPInterface.c:135 Posted read
^C
c:\Users\Vadym\Desktop\cjdns>

**Comment by interfect** [comment]
So the problem is that cjdns is not connecting? How do you know that cjdns isn't connecting? (Maybe the messages you are seeing are just warnings and cjdns is actually working properly despite printing lots of output.)

What does the tun interface look like while this is going on, in the `ipconfig /all` output? Is it getting an `fcXX::...` address?

If it's just that the connectivity test shipped with the installer fails, maybe the public peers the installer points you at are dead. Have you tried adding some other peers?

**Comment by interfect** [comment]
I notice in your log that you don't have anything about connecting or trying to connect to peers. Do you have any peers added in your config file?

**Comment by vikulin** [comment]
yes, I did. ipconfig shows successfully created virtual IPv6 interface

**Comment by vikulin** [comment]
" Message with unrecognized address from TUN" message blocks the app normal work meaning that
`static Iface_DEFUN incomingFromTun(struct Message* message, struct Iface* tunIf)` method does not return a connection.

**Comment by interfect** [comment]
Can you be more specific than "blocks the app normal work"? What test,
specifically, did you do to see if it was working? What was the expected
behavior? What was the observed behavior?

It looks like that message is coming from the IpTunnel code.

Are you trying to set up an IpTunnel? Is the node in question the server or
the client for the tunnel? Do you control both endpoints? What does the
IpTunnel config look like?

**Comment by vikulin** [comment]
I did simple attempt to connect using a valid peer checking it on Windows. Also I did cross-build of the binary as recommended in your repo.
1. Added 100% valid peer (checked on ubuntu at that time)
2. Ran cjdroute --nobg < cjdroute.conf as shown in log

AR: no ping -6 works
ER: ping -6 should work on fc02:2735:e595:bb70:8ffc:5293:8af8:c4b7
(https://www.fc00.org/#fc02:2735:e595:bb70:8ffc:5293:8af8:c4b7)
